python/cuml/cuml/internals/array.py

```diff
@@ -707,8 +707,15 @@ def to_output(
                 out_index = cudf_to_pandas(self.index)
             else:
                 out_index = self.index
+            if output_mem_type.is_device_accessible:
+                # Do not convert NaNs to nulls in cuDF
+                df_kwargs = {"nan_as_null": False}
```

Comment:
The `nan_as_null` argument is [deprecated for pandas](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.__dataframe__.html), are you planning a similar deprecation in cuDF? Should we add some kind of comment regarding this?

Reply:
The `__dataframe__` API is different from the regular constructor (`__init__`, https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.html). So cuDF doesn't necessarily need to align with pandas's `__dataframe__` API 

(For context, DataFrame libraries are moving away from `__dataframe__`. cuDF recently removed support for it https://github.com/rapidsai/cudf/pull/17909)